### DISCUSSION TITLE
How to invite users to Vaultwarden instance without using Admin UI?

### DISCUSSION THREAD

**Discussion by jayknyn** [dani-garcia/vaultwarden#3591] [Q&A] [answered]
Hi, I searched and wasn't able to find any similar topics on this.

Currently when I want to invite someone to my Vaultwarden instance (not to my organization) so that they can setup their own organization, I use the invite user form field in the Admin UI. 

Is there a way to send this invite user email without using the Admin GUI, like via an API call or a command from the command line? I know that the CLI client won't work because that only interacts with my personal user profile (my orgs).

![Screen Shot 2023-06-16 at 08 51 16](https://github.com/dani-garcia/vaultwarden/assets/45739598/56953842-01a7-493c-9e35-c3d5497a860b)

**Comment by tessus** [comment, marked as answer]
Indeed there is. [code](https://github.com/dani-garcia/vaultwarden/blob/5b7d7390b0cc26ebde96d1b8a835e384ee8deb47/src/api/admin.rs#L279-L302
) 

So, you should be able to send a `POST` request to `/admin/invite` with a json payload:

```json
{ "email": "user_to_be@invited.com" }
```

**Reply by jayknyn** [reply]
That's great. Thanks so much, Helmut!

**Reply by tessus** [reply]
You are welcome.

**Comment by Gecko-with-a-hat** [comment]
Apologies for reviving an old thread, but I’m struggling to figure out how to implement the solution described above.

I’ve posted my question on GitHub, Reddit, and the Vaultwarden forum, but haven’t received any helpful responses. I’ve also tried using ChatGPT and going through the [code](https://github.com/dani-garcia/vaultwarden/blob/5b7d7390b0cc26ebde96d1b8a835e384ee8deb47/src/api/admin.rs#L279-L302) to understand how to invite new users via the API, but it still doesn’t seem to work.

Here’s the GitHub issue I posted: https://github.com/dani-garcia/vaultwarden/discussions/5054. I’ve shared my Python code there, but I can’t seem to authenticate with the server. Could @jayknyn or @tessus please take a look and let me know what I’m doing wrong?
